File: api/SpeechViewer/SpeechesByAuthor.cs
namespace SpeechViewer
{
    /// <summary>
    /// Basic information about an author and speeches that they wrote.
    /// </summary>
    class SpeechesByAuthor
    {
        /// <summary>
        /// The author of the speeches.
        /// </summary>
        public AuthorInfo Author { get; set; }

        /// <summary>
        /// Speeches written by the author.
        /// </summary>
        public AuthorSpeechInfo[] Speeches { get; set; }
    }
}

File: web/src/speeches-by-author.ts
import { AuthorInfo } from './author-info';
import { AuthorSpeechInfo } from './author-speech-info';

/**
 * Basic information about an author and speeches that they wrote.
 */
export interface SpeechesByAuthor {
  /**
   * The author of the speeches.
   */
  author: AuthorInfo;

  /**
   * Speeches written by the author.
   */
  speeches: AuthorSpeechInfo[];
}
